src/NServiceBus.Metrics.PerformanceCounters/PerformanceCounterUpdater.cs

```diff
@@ -1,38 +1,37 @@
 ﻿using System.Collections.Generic;
-using Newtonsoft.Json.Linq;
+using System.Linq;
+using NServiceBus;
 
 class PerformanceCounterUpdater
 {
-    public PerformanceCounterUpdater(PerformanceCountersCache cache, Dictionary<string, CounterInstanceName?> legacyInstanceNameMap)
+    public PerformanceCounterUpdater(PerformanceCountersCache cache, Dictionary<string, CounterInstanceName?> legacyInstanceNameMap, string endpointName)
     {
         this.legacyInstanceNameMap = legacyInstanceNameMap;
+        this.endpointName = endpointName;
         this.cache = cache;
     }
 
-    public void Update(string payload)
+    public void Observe(ProbeContext context)
     {
-        var rootObject = JObject.Parse(payload);
-
-        var context = rootObject.Value<string>("Context");
-        var meters = rootObject["Meters"]?.ToObject<List<Meter>>() ?? new List<Meter>();
-        foreach (var meter in meters)
+        context.Signals.ToList().ForEach(sp =>
         {
             CounterInstanceName? instanceName;
-            legacyInstanceNameMap.TryGetValue(meter.Name, out instanceName);
+            legacyInstanceNameMap.TryGetValue(sp.Name, out instanceName);
+
+            var performanceCounterInstance = cache.Get(instanceName ?? new CounterInstanceName(sp.Name, endpointName));
 
-            var performanceCounterInstance = cache.Get(instanceName ?? new CounterInstanceName(meter.Name, context));
-            performanceCounterInstance.RawValue = meter.Count;
-        }
+            sp.Register(() => performanceCounterInstance.Increment());
```

Comment:
This allocations is done only once, correct? The value of removing it is a direct call on the passed object rather than having a closure class that contains it, loads and then calls. Is that right? Is there anything else?